### PR TITLE
Feature/dmp 641 Market on google maps

### PR DESCRIPTION
**Pull request by dianabasilio**
## What is done

[[DMP-641] Market on google maps](https://deacero.atlassian.net/browse/DMP-641)

I think this pr also covers DMP-642

[[DMP-642] Market on google maps](https://deacero.atlassian.net/browse/DMP-642)

## How is done

- Added functionality so that  map location changes depending on postal code.
- Added Marker to maps.
image

- Zoom updates.

## How to test

1. Run the project via `npm start`
2. Go to registrarse.
3. Fill first form.
4. On the second form map should be displayed and you should be able to see and change the marker with no problems.

### CODE REVIEW

File: src/config/api/api.ts [hunk 1]
```diff
@@ -61,6 +61,10 @@ const getProductDetail = (productId: number): Promise<ProductDetail> => {
   return fetchData(API.productDetail(productId));
 };
 
+const getLatLngFromPostalCode = (postalCode: number, country: string, apiKey: string): Promise<any> => {
+  return fetchData(`https://maps.googleapis.com/maps/api/geocode/json?skip_auth&components=postal_code:${postalCode}|country:${country}&region=AU&key=${apiKey}`);
```
Comment (markriverac) [inline src/config/api/api.ts:65]: This api route should be defined just as the other routes we use. In the api.types file

File: src/components/common/Map/Map.tsx [hunk 2]
```diff
@@ -24,14 +31,19 @@ const Map: React.FC<MapProps> = ({ zoom = 12, center = sanPedroLocation }) => {
...
     return (
       <GoogleMap
         zoom={zoom}
         center={centered}
         mapContainerStyle={containerStyle}
+        onClick={mapClicked && mapClicked}
+        onZoomChanged={handleZoomChanged}
       >
-        <Marker position={centered} />
+        <MarkerF position={centered} />
```
Comment (markriverac) [inline src/components/common/Map/Map.tsx:46]: why the rename or what F means?

**Review by markriverac — CHANGES_REQUESTED** [review]

File: src/components/common/Map/Map.tsx (hunk 2 above)
Comment (dianabasilio) [reply src/components/common/Map/Map.tsx:46]: turns out Marker does not work anymore on recent react versions, the correct name now for this is MarkerF.

File: src/config/api/api.ts (hunk 1 above)
Comment (markriverac) [reply src/config/api/api.ts:66]: @dianabasilio what I meant with my comment is that we shouldn't do a `fetchData("maps.googleapis...");`
We should do a `fetchData(API.mapsAPI(postalCode, country, apiKey)``
Also, I think the apiKey should be somewhere defined, and not to be called as a param

**Review by markriverac — CHANGES_REQUESTED** [review]

File: src/components/Form/MapField/MapField.tsx [hunk 3]
```diff
@@ -16,11 +16,7 @@ const MapField: React.FC<OptionsField> = ({ label }) => {
   useEffect(() => {
     zipCode &&
       zipCode.length === 5 &&
-      getLatLngFromPostalCode(
-        parseInt(zipCode),
-        "MX",
-        process.env.REACT_APP_GOOGLE_MAPS_API_KEY!
-      ).then((data) => {
+      getLatLngFromPostalCode(parseInt(zipCode), "MX").then((data) => {
```
Comment (markriverac) [inline src/components/Form/MapField/MapField.tsx:19]: Nice! My comment didn't cover this change, thank you!

File: src/config/api/api.ts [hunk 4]
```diff
@@ -62,8 +62,11 @@ const getProductDetail = (productId: number): Promise<ProductDetail> => {
   return fetchData(API.productDetail(productId));
 };
 
-const getLatLngFromPostalCode = (postalCode: number, country: string, apiKey: string): Promise<CodePostalInfo> => {
-  return fetchData(`https://maps.googleapis.com/maps/api/geocode/json?skip_auth&components=postal_code:${postalCode}|country:${country}&region=AU&key=${apiKey}`);
+const getLatLngFromPostalCode = (
+  postalCode: number,
+  country: string
+): Promise<CodePostalInfo> => {
+  return fetchData(API.mapsAPI(postalCode, country));
```
Comment (markriverac) [inline src/config/api/api.ts:69]: 😮‍💨

**Review by markriverac — APPROVED** [review]

**Pull request merged**